### PR TITLE
Minutes from 5 December Software Forum meeting

### PR DESCRIPTION
**Pull request by graeme-a-stewart**
@JavierCVilla and @JulienPeloton might also want to take a look,

### CODE REVIEW

File: organization/_posts/2018/2018-12-05-softwareforum.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+-   Performance test was for \~small data elements (5 doubles); total
+    size 100GB.
+-   The schema for the dataset is fixed for each case; but see issues when
+    many input files are used (causing the metadata bottleneck).
+-   3D spatial data not well supported in Spark ecosystem.
+    -   Most packages work with 2D data.
+-   Re-partitioning in space is only one option - in general want to be
+    able to repartition on arbitrary metrics:
+    -   Need to build partition and then shuffle the data (2 phases).
+    -   USF = User Defined Function - this is difficult to optimise as
```
Comment (JulienPeloton) [inline organization/_posts/2018/2018-12-05-softwareforum.md]: typo: UDF.

**Comment by JulienPeloton** [comment]
Thanks @graeme-a-stewart for compiling these notes! It looks good to me.

File: organization/_posts/2018/2018-12-05-softwareforum.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        phase can begin.
+    -   Problem in HEP as events are inhomogeneous - hard to predict
+        exactly how long processing will take.
+-   Observation that analysing ROOT files from HDFS is slower than with
+    EOS.
+-   Want to avoid too much up-front tuning. Can you apply a task
+    stealing approach, idle workers try to grab work from other busy
+    tasks?
+    -   Problem is that Spark is doing this part and it doesn’t know how
+        to break apart the tasks.
```
Comment (JavierCVilla) [inline organization/_posts/2018/2018-12-05-softwareforum.md]: I would add here:

> however, final results can be improved by choosing the correct granularity of the tasks.

File: organization/_posts/2018/2018-12-05-softwareforum.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        structures that were not known at data loading time then the
+        map/reduce workflow performs poorly when the output is as
+        large as the input.
+    -   Q. Problem with the memory usage? Is the algorithm optimal?
+        - Default Spark partitioning is not doing so well, hence
+          the O(N^2) scaling seen.
+
+Distributed data analysis with ROOT RDataFrame and Spark
+--------------------------------------------------------
+-   Working with ROOT to establish working model for analysis for FCC
```
Comment (JavierCVilla) [inline organization/_posts/2018/2018-12-05-softwareforum.md]: I'm a bit confused here, I mentioned FCC to present myself however this project is not related to FCC at the moment.

**Comment by graeme-a-stewart** [comment]
Thanks both - should have addressed all of those points now.
